crates/aptos/src/move_tool/disassembler.rs

```diff
@@ -0,0 +1,131 @@
+use async_trait::async_trait;
+use move_bytecode_source_map::{mapping::SourceMapping, utils::source_map_from_file};
+use move_ir_types::location::Spanned;
+
+/// Disassemble the Move bytecode pointed to
+#[derive(Debug, Parser)]
+pub struct AptosDisassembler {
+    /// Skip printing of private functions.
+    #[clap(long = "skip-private")]
+    pub skip_private: bool,
+
+    /// Do not print the disassembled bytecodes of each function.
+    #[clap(long = "skip-code")]
+    pub skip_code: bool,
+
+    /// Do not print locals of each function.
+    #[clap(long = "skip-locals")]
+    pub skip_locals: bool,
+
+    /// Do not print the basic blocks of each function.
+    #[clap(long = "skip-basic-blocks")]
+    pub skip_basic_blocks: bool,
+
+    /// Treat input file as a script (default is to treat file as a module)
+    #[clap(short = 's', long = "script")]
+    pub is_script: bool,
+
+    /// The path to the bytecode file to disassemble; let's call it file.mv. We assume that two
+    /// other files reside under the same directory: a source map file.mvsm (possibly) and the Move
+    /// source code file.move.
+    #[clap(short = 'b', long = "bytecode")]
+    pub bytecode_file_path: String,
+
+    /// (Optional) Path to a coverage file for the VM in order to print trace information in the
+    /// disassembled output.
+    #[clap(short = 'c', long = "move-coverage-path")]
+    pub code_coverage_path: Option<String>,
+}
+
+#[async_trait]
+impl CliCommand<String> for AptosDisassembler {
+    fn command_name(&self) -> &'static str {
+        "AptosDisassembler"
+    }
+
+    async fn execute(self) -> CliTypedResult<String> {
+        let move_extension = MOVE_EXTENSION;
+        let mv_bytecode_extension = MOVE_COMPILED_EXTENSION;
+        let source_map_extension = SOURCE_MAP_EXTENSION;
+
+        let source_path = Path::new(&self.bytecode_file_path);
+        let extension = source_path
+            .extension()
+            .expect("Missing file extension for bytecode file");
+        if extension != mv_bytecode_extension {
+            return Ok(format!(
+                "Bad source file extension {:?}; expected {}",
+                extension, mv_bytecode_extension)
+            );
+        }
+
+        let bytecode_bytes = fs::read(&self.bytecode_file_path).expect("Unable to read bytecode file");
+
+        let source_path = Path::new(&self.bytecode_file_path).with_extension(move_extension);
+        let source = fs::read_to_string(&source_path).ok();
+        let source_map = source_map_from_file(
+            &Path::new(&self.bytecode_file_path).with_extension(source_map_extension),
+        );
+
+        let mut disassembler_options = DisassemblerOptions::new();
+        disassembler_options.print_code = !self.skip_code;
+        disassembler_options.only_externally_visible = self.skip_private;
+        disassembler_options.print_basic_blocks = !self.skip_basic_blocks;
+        disassembler_options.print_locals = !self.skip_locals;
```

Comment:
The fields of `DisassemblerOptions` are all pub, let's just build the struct normally:
```suggestion
        let disassembler_options = DisassemblerOptions {
            print_code: !self.skip_code,
            only_externally_visible: self.skip_private,
            print_basic_blocks: !self.skip_basic_blocks,
            print_locals: !self.skip_locals,
        };
```